src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1818,6 +1834,9 @@ RouterSelectTask(Query *originalQuery, RelationRestrictionContext *restrictionCo
 	uint64 shardId = INVALID_SHARD_ID;
 	List *selectShardList = NIL;
 
+	/* router planner should create task even if it deosn't hit a shard at all */
+	restrictionContext->replaceRTEsWithNull = true;
```

Comment:
It seems pretty wrong to use the restrictionContext to store this information. We'll probably end up using them in other places too, and there replaceRTEsWithNull doesn't seem to make much sense.